Runtime/Runner/Scenes/LanguageSelectorController.cs

```diff
@@ -0,0 +1,240 @@
+    private static Dictionary<string, string> myDictionary;
+    private static Dictionary<string, string> defaultDictionary;
+    private static string Language;
+
+    private void Awake()
+    {
+        if (instance == null)
+            instance = this;
+        else if (instance != this)
+            Destroy(gameObject);
+        SetActive(true);
+        GetLanguages();
+        RefreshLanguageList();
+        SetUpJSONFiles(true);
+        FillDictionary(true);
+        DontDestroyOnLoad(gameObject);
+    }
+
+    public void Back()
+    {
+        GameObject.DestroyImmediate(this.gameObject);
+    }
+
+    public string GetLanguageFromTitle(string title) {
+        foreach(string languageCode in languages.Keys) {
+            if(languages[languageCode] == title) {
+                return languageCode;
+            }
+        }
+        return null;
+    }
+
+    public void SetLanguageFromTitle(string title) {
+        string  lang = GetLanguageFromTitle(title);
+        if (lang == null) {
+            Debug.LogError("Language not found. (" + title + ")");
+        } else {
+            Language = title;
+        }
+    }
+
+    //Selects a language by flag button in Title scene
+    public void fillDictionaryAndRunLoginScene()
+    {
+        SetUpJSONFiles(false);
+        FillDictionary(false);
+        SimvaPlugin.Instance.RunScene("Simva.Login");
+    }
+
+      //Selects a language by flag button in Title scene
+    public void SetLanguageCode(string code)
+    {
+        Language = code;
+    }
+
+    public void SetActive(bool active)
+    {
+        this.gameObject.SetActive(active);
+    }
+
+    public string GetCurrentLanguage() { return Language; }
+
+    public void RefreshLanguageList()
+    {
+        // Clear existing children
+        if(languageGridLayout) {
+            foreach (Transform child in languageGridLayout.transform)
+            {
+                Destroy(child.gameObject);
+            }
+        }
+
+        // Spawn one UI item per selected language
+        foreach (string lang in SimvaPlugin.Instance.SelectedLanguages)
+        {
+            GameObject item = Instantiate(languageItemPrefab, languageGridLayout.transform);
+            foreach(string languageCode in languages.Keys) {
+                if(languages[languageCode] == lang) {
+                    item.name = languageCode;
+                    item.SetActive(true);
+                    continue;
+                }
+            }
+        }
+    }
+
+    void GetLanguages()
+    {
+        languages = new Dictionary<string, string>();
+
+        TextAsset[] filler = Resources.LoadAll<TextAsset>("Localization");
+        foreach (TextAsset obj in filler) {
+            if (obj.name == "lang") {
+                JObject jObject = JObject.Parse(obj.text);
+                var code="";
+                var name="";
+                var custom=false;
+                foreach (var entry in jObject) {
```

Comment:
Minimal performance issue. Optional to fix by checking if jObject contains the element named code, displayName and custom instead should be optimal.